src/main/java/hudson/remoting/Channel.java

```diff
@@ -494,6 +519,8 @@ protected Channel(ChannelBuilder settings, CommandTransport transport) throws IO
 
         transport.setup(this, new CommandReceiver() {
             public void handle(Command cmd) {
+                commandsReceived++;
```

Comment:
No, commands are received in serial order, so it is not possible for this to happen in different threads that do not have memory barrier established.